### PR TITLE
Add `.A[]` access for table view support in debug

### PR DESCRIPTION
**Pull request by twicki**
Add the `GlobalTable` feature support for `debug` backend.

Global table is an "off-grid" array that can only be accessed absolutely via data dimensions.

### CODE REVIEW

File: tests/cartesian_tests/integration_tests/multi_feature_tests/test_debug_backend.py [hunk 1]
```diff
@@ -350,3 +350,26 @@ def test_stencil(
...
+
+
+def test_table_access_stencil():
+    table_view = np.ones((4,), dtype=np.float64)
+    field_out = gt_storage.zeros(
+        dtype=np.float64, backend="debug", shape=(4, 4, 4), aligned_index=(0, 0, 0)
+    )
+    table_view[:] = [2, 3, 4, 5]
+
+    @gtscript.stencil(backend="debug")
```
Comment (romanc) [inline tests/cartesian_tests/integration_tests/multi_feature_tests/test_debug_backend.py:362]: random thought: once we bring the debug backend back into gt4py mainline, we should check which tests are specific to the debug backend and which tests should actually be annotated with `@gtscript.stencil(backend=ALL_BACKENDS`  (with potential excludes referencing issues with missing implementation in other backends)

File: src/gt4py/cartesian/gtc/debug/debug_codegen.py [hunk 2]
```diff
@@ -247,7 +247,10 @@ def visit_FieldAccess(self, field_access: FieldAccess, **_) -> str:
             data_index_access = ",".join(
                 [self.visit(data_index) for data_index in field_access.data_index]
             )
-            full_string = field_access.name + "[" + offset_str + "," + data_index_access + "]"
+            if offset_str == "":
+                full_string = field_access.name + f"[{data_index_access}]"
+            else:
+                full_string = field_access.name + "[" + offset_str + "," + data_index_access + "]"
```
Comment (romanc) [inline src/gt4py/cartesian/gtc/debug/debug_codegen.py]: How about we are a bit more pythonic?
```suggestion
            index_str = f"{offset_str},{data_index_access}" if offset_str else data_index_access
            full_string = f"{field_access.name}[{index_str}]"
```

**Review by romanc — APPROVED** [review]
Please fix the PR description. It's nice to have some context.

File: tests/cartesian_tests/integration_tests/multi_feature_tests/test_debug_backend.py (hunk 1 above)
Comment (FlorianDeconinck) [reply tests/cartesian_tests/integration_tests/multi_feature_tests/test_debug_backend.py:362]: I'll log that in right away, I can feel myself forgetting

Comment (FlorianDeconinck) [reply tests/cartesian_tests/integration_tests/multi_feature_tests/test_debug_backend.py:362]: https://github.com/GEOS-ESM/SMT-Nebulae/issues/98

**Review by FlorianDeconinck — APPROVED** [review]

**Pull request merged**